base/compiler/typeinfer.jl

```diff
@@ -627,19 +630,20 @@ function return_type(@nospecialize(f), @nospecialize(t))
     return ccall(:jl_call_in_typeinf_world, Any, (Ptr{Ptr{Cvoid}}, Cint), Any[_return_type, f, t, world], 4)
 end
 
-function _return_type(@nospecialize(f), @nospecialize(t), world)
-    params = Params(world)
+_return_type(@nospecialize(f), @nospecialize(t), world) = _return_type(NativeInterpreter(), f, t, world)
+
+function _return_type(interp::AbstractInterpreter, @nospecialize(f), @nospecialize(t), world)
```

Comment:
I think method lookup is sort of a function of the interpreter, so I'd rather not pass the world through directly. Instead, I think it'd be fair to have: `world(interp::NativeInterpreter) = interp.world`
I think eventually we may want to factor the method lookup logic out of this function separately, but that doesn't have to be now.

Reply:
I already have a `get_world_counter(interp)` so that should be easy enough; I saw that the `world` was being explicitly passed previously, and I didn't want to change that since I wasn't sure if the `world` could ever be different than the world embedded in the `interp` or not.